go-controller/pkg/factory/factory.go

```diff
@@ -423,27 +443,37 @@ const (
 )
 
 var (
-	podType       reflect.Type = reflect.TypeOf(&kapi.Pod{})
-	serviceType   reflect.Type = reflect.TypeOf(&kapi.Service{})
-	endpointsType reflect.Type = reflect.TypeOf(&kapi.Endpoints{})
-	policyType    reflect.Type = reflect.TypeOf(&knet.NetworkPolicy{})
-	namespaceType reflect.Type = reflect.TypeOf(&kapi.Namespace{})
-	nodeType      reflect.Type = reflect.TypeOf(&kapi.Node{})
+	podType            reflect.Type = reflect.TypeOf(&kapi.Pod{})
+	serviceType        reflect.Type = reflect.TypeOf(&kapi.Service{})
+	endpointsType      reflect.Type = reflect.TypeOf(&kapi.Endpoints{})
+	policyType         reflect.Type = reflect.TypeOf(&knet.NetworkPolicy{})
+	namespaceType      reflect.Type = reflect.TypeOf(&kapi.Namespace{})
+	nodeType           reflect.Type = reflect.TypeOf(&kapi.Node{})
+	egressFirewallType reflect.Type = reflect.TypeOf(&egressfirewallapi.EgressFirewall{})
+	crdType            reflect.Type = reflect.TypeOf(&apiextensionsapi.CustomResourceDefinition{})
 )
 
 // NewWatchFactory initializes a new watch factory
-func NewWatchFactory(c kubernetes.Interface) (*WatchFactory, error) {
+func NewWatchFactory(c kubernetes.Interface, ec egressfirewallclientset.Interface, crd apiextensionsclientset.Interface) (*WatchFactory, error) {
 	// resync time is 12 hours, none of the resources being watched in ovn-kubernetes have
 	// any race condition where a resync may be required e.g. cni executable on node watching for
 	// events on pods and assuming that an 'ADD' event will contain the annotations put in by
 	// ovnkube master (currently, it is just a 'get' loop)
 	// the downside of making it tight (like 10 minutes) is needless spinning on all resources
 	wf := &WatchFactory{
-		iFactory:  informerfactory.NewSharedInformerFactory(c, resyncInterval),
-		informers: make(map[reflect.Type]*informer),
-		stopChan:  make(chan struct{}),
+		iFactory:    informerfactory.NewSharedInformerFactory(c, resyncInterval),
+		efClientset: ec,
+		crdFactory:  apiextensionsinformerfactory.NewSharedInformerFactory(crd, resyncInterval),
+		informers:   make(map[reflect.Type]*informer),
+		stopChan:    make(chan struct{}),
 	}
 	var err error
+
+	err = apiextensionsapi.AddToScheme(apiextensionsscheme.Scheme)
```

Comment:
consider: should the crd code be factored out into it's own `InitializeCRDFactory`?

Reply:
Since the CRD is always present it can be in the generic NewWatchFactory

Reply:
Since the CRD should always be present I don't think there is a case to be made to spin it into it's own function.

Reply:
That's fine.